src/ageml/modelling.py

```diff
@@ -367,105 +372,132 @@ def fit_age(self, X, y):
-        if self.hyperparameter_grid != {}:
-            print("Running Hyperparameter optimization...")
-            opt_pipeline = model_selection.GridSearchCV(self.pipeline,
-                                                        self.hyperparameter_grid,
-                                                        cv=self.CV_split,
-                                                        scoring="neg_mean_absolute_error")
-            opt_pipeline.fit(X, y)
-            print(f"Hyperoptimization best parameters: {opt_pipeline.best_params_}")
-            # Set best parameters in pipeline
-            self.pipeline.set_params(**opt_pipeline.best_params_)
+        if self.model_type != "hyperopt":
+            # Optimize hyperparameters if required
+            if self.hyperparameter_tuning > 1:
+                print("Running Hyperparameter optimization...")
+            else:
+                print("No hyperparameter optimization will be done.")
+            # Generate the hyperparameter grid. If {}, default values are used
+            hyperparameter_grid = model_selection.ParameterGrid(self.hyperparameter_grid)
+            pipelines = []
+            # Generate pipelines with different hyperparameters from the grid
+            for grid_point in hyperparameter_grid:
+                original_pipeline = copy.deepcopy(self.pipeline)
+                point_pipeline = original_pipeline.set_params(**grid_point)
+                pipelines.append(point_pipeline)
+            # Variables of interest
+            pred_age = np.zeros(y.shape[0])
+            corrected_age = np.zeros(y.shape[0])
+            best_split_mae = 1e10
+            mae_means_test = []
+            # Loop through the pipelines, and then loop through the CV splits
+            for cv_pipeline in pipelines:
+                print(f"\nRunning CV splits with pipeline:\n{cv_pipeline}")
+                temp_pred_age = np.zeros(y.shape[0])
+                temp_corr_age = np.zeros(y.shape[0])
+                split_metrics_train = []
+                split_metrics_test = []
+                kf_hyperopt = model_selection.KFold(n_splits=self.CV_split, random_state=self.seed, shuffle=True)
+                for i, (train, test) in enumerate(kf_hyperopt.split(X)):
+                    X_train, X_test = X[train], X[test]
+                    y_train, y_test = y[train], y[test]
+
+                    # Train model with the hyperparameters
+                    cv_pipeline.fit(X_train, y_train)
+
+                    # Predictions on train and test set
+                    y_pred_train = cv_pipeline.predict(X_train)
+                    y_pred_test = cv_pipeline.predict(X_test)
+
+                    # Metrics in Train and Test sets
+                    split_metrics_train.append(self.calculate_metrics(y_train, y_pred_train))
+                    split_metrics_test.append(self.calculate_metrics(y_test, y_pred_test))
+
+                    # Print metrics
```

Comment:
Why are you priniting for each hypeparemeter optimization the train and test folds? Should only be printed for final best pipeline. Store this information and print at the end.